arch: arc: enable divide zero exception

IRISZZW opened the pull request:
STATUS32.DZ(bit 13) is the EV_DivZero exception enable bit, and it's
not enabled by default. we need to set it explicitly to enable divide
zero exception on early boot and each thread's setup.

The DZ bit is ignored on write and read as zero when there is no
hardware division configured. So we can simply set DZ bit even if
there is no hardware division configured.

Fix: #33459

ruuddw commented on arch/arc/core/thread.c:
```diff
@@ -178,6 +178,7 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 #ifdef CONFIG_ARC_USE_UNALIGNED_MEM_ACCESS
 	iframe->status32 |= _ARC_V2_STATUS32_AD;
 #endif
+	iframe->status32 |= _ARC_V2_STATUS32_DZ;
```
Why not add this to the initial assignments above, as there is no conditional involved.

IRISZZW replied:
Like this ?

```
#ifdef CONFIG_USERSPACE
	/* enable US bit, US is read as zero in user mode. This will allow user
	 * mode sleep instructions, and it enables a form of denial-of-service
	 * attack by putting the processor in sleep mode, but since interrupt
	 * level/mask can't be set from user space that's not worse than
	 * executing a loop without yielding.
	 */
	iframe->status32 = _ARC_V2_STATUS32_US | _ARC_V2_STATUS32_DZ;
	if (is_user(thread)) {
		iframe->pc = (uint32_t)z_user_thread_entry_wrapper;
	} else {
		iframe->pc = (uint32_t)z_thread_entry_wrapper;
	}
#else
	iframe->status32 = _ARC_V2_STATUS32_DZ;
	iframe->pc = ((uint32_t)z_thread_entry_wrapper);
#endif /* CONFIG_USERSPACE */
```

ruuddw replied:
Yes, that's what I had in mind.

IRISZZW committed:
arch: arc: enable divide zero exception

STATUS32.DZ(bit 13) is the EV_DivZero exception enable bit, and it's
not enabled by default. we need to set it explicitly to enable divide
zero exception on early boot and each thread's setup.

The DZ bit is ignored on write and read as zero when there is no
hardware division configured. So we can simply set DZ bit even if
there is no hardware division configured.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

evgeniy-paltsev commented:
@nashif do we have any politics in Zephyr about the minimum time before the pull request get merged?

I have significant notes about this, but it has been already merged :/
